File: HW9Assigned/low8/MakeShared/MallardDuck.cpp
#include <iostream>
#include "MallardDuck.h"
#include "Quack.h"
#include "FlyWithWings.h"

MallardDuck::MallardDuck( ) {
   quackBehavior = new Quack( );
   flyBehavior = new FlyWithWings( );
}

void MallardDuck::display( ) {
   std::cout << "I’m a real mallard duck" << std::endl;
}

File: HW14Assigned/low8/duck/MallardDuck.java
package low8.duck;
import low8.fly.*;
import low8.quack.*;
import low8.eggs.*;

public class MallardDuck extends Duck {
   public MallardDuck( ) {
      quackBehavior = new Quack( );
      flyBehavior = new FlyWithWings( );
      laysEggsBehavior = new LaysEggsBroody( );
   }
   
   public void display( ) {
      System.out.println("I’m a real mallard duck");
   }
}
